base/strings/substring.jl

```diff
@@ -47,6 +47,11 @@ end
 SubString(s::AbstractString) = SubString(s, 1, lastindex(s))
 SubString{T}(s::T) where {T<:AbstractString} = SubString{T}(s, 1, lastindex(s))
 
+
+@propagate_inbounds view(s::AbstractString, r) = SubString(s, r)
+@propagate_inbounds maybeview(s::AbstractString, r::UnitRange{<:Integer}) = view(s, r)
+@propagate_inbounds maybeview(s::AbstractString, args...) = getindex(s, args...)
```

Comment:
`maybeview` is required for `@views` to work well on a large block that includes somethings we can take views of and others that we can't.

Its actually really nice as a setup, i was worried about this, and until i realized, I had added extra non-SubString overloads to `view`